### DISCUSSION TITLE
Understanding the serverless execution timeout

### DISCUSSION THREAD

**Discussion by alexbjorlig** [vercel/community#2772] [Help] [answered]
### Summary

I'm new to the serverless platform.
If I write a serverless function, that returns a HTTP answer within the given limit, can the function continue to run? And for how long?

We are currently on the PRO plan

### Steps to Reproduce

-

**Comment by swarnava** [comment, marked as answer]
Serverless Functions on Vercel enforce a maximum execution timeout. This means that the function must respond to an incoming HTTP request before the timeout has been reached.

If the Serverless Function does not respond to the HTTP request within the timeout, then a 504 error status code is returned with the error code [FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/error/application/FUNCTION_INVOCATION_TIMEOUT).

The timeout is determined by the ["Serverless Function Execution Timeout (Seconds)"](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) limit, which is based on the plan that the Personal Account, or Team has enabled when creating the deployment. Since you are on Pro plan, the limit will be 60 seconds.

**Reply by janacm** [reply]
Default now is 15s even on the pro plan.

**Comment by alexbjorlig** [comment]
Ok, that makes sense.
But what if the function responds to an incoming HTTP request straight away, and then starts a 

```js
setInterval(() => {
  console.log("Im alive!")
}, 1000)
```

For how long would that keep running?

**Reply by swarnava** [reply]
It will run for 60 seconds and then timeout with 504 error

**Reply by alexbjorlig** [reply]
Ok. It's just interesting how the documentation focuses on `must respond to the HTTP request`, and does not talk about total execution time. But I guess the case is settled then 👍

**Reply by janacm** [reply]
Depends on if this is an edge function or a serverless function - edge doesn't have execution limits as long as you respond within 25 seconds
